# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IndexedDatabase.h
# Commit: ad7b08acadd6
# Full Hash: ad7b08acadd691fc86b52ba7f8babfaa57972c56
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-03-19 17:04:52
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefPtr for FileInfo. r=dom-workers-and-storage-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65550
# ==============================================================================

diff -r bb094a6084b3 -r ad7b08acadd6 dom/indexedDB/IndexedDatabase.h
--- a/dom/indexedDB/IndexedDatabase.h	Thu Mar 19 09:32:53 2020 +0000
+++ b/dom/indexedDB/IndexedDatabase.h	Thu Mar 19 09:33:31 2020 +0000
@@ -13,6 +13,7 @@
 #include "nsTArray.h"
 #include "InitializedOnce.h"
 #include "FileInfoFwd.h"
+#include "SafeRefPtr.h"
 
 namespace mozilla {
 namespace dom {
@@ -110,7 +111,7 @@
   StructuredCloneFileParent& operator=(StructuredCloneFileParent&&) = delete;
 
   // In IndexedDatabaseInlines.h
-  StructuredCloneFileParent(FileType aType, RefPtr<FileInfo> aFileInfo);
+  StructuredCloneFileParent(FileType aType, SafeRefPtr<FileInfo> aFileInfo);
 
   // In IndexedDatabaseInlines.h
   ~StructuredCloneFileParent();
@@ -123,10 +124,10 @@
   const indexedDB::FileInfo& FileInfo() const { return ***mContents; }
 
   // In IndexedDatabaseInlines.h
-  RefPtr<indexedDB::FileInfo> FileInfoPtr() const;
+  SafeRefPtr<indexedDB::FileInfo> FileInfoPtr() const;
 
  private:
-  InitializedOnce<const Maybe<RefPtr<indexedDB::FileInfo>>> mContents;
+  InitializedOnce<const Maybe<SafeRefPtr<indexedDB::FileInfo>>> mContents;
 };
 
 struct StructuredCloneReadInfoBase {